src/app/app-utils.ts

```diff
@@ -66,14 +66,62 @@ export class AppUtils {
    * Set the active property on the navigation items.
    */
   public static setActiveLink(items: NavItemsModel[], currentUrl: string): NavItemsModel[] {
+    let fullUrl = false;
+    let matchingUrl = '';
+    [fullUrl, matchingUrl] = AppUtils.checkTabs(items, currentUrl);
     return items.map(item => {
       return {
         ...item,
-        active: item.href === currentUrl
+        active: fullUrl ? item.href === currentUrl : item.href === matchingUrl
       };
     });
   }
 
+  /**
+   * Tab logic - Works out which tab needs to be selected via the current tab urls and the given url
+   *
+   * @param items - the tab urls
+   * @param currrentUrl - the url being tested
+   * @return - a list including boolean stating whether the full url is given or the similar matching url
+   */
+  public static checkTabs(items: NavItemsModel[], currentUrl: string): any[] {
+    let fullUrl = false;
+    let maxLength = 0;
+    let matchingUrl = '';
+    let checkHrefs = [];
+    for (const checkItem of items) {
+      // allow checking of /tasks urls properly
+      checkHrefs = checkItem.href === '/tasks' ? ['/tasks/list', '/tasks/available'] : [checkItem.href];
+      fullUrl = AppUtils.isFullUrl(checkItem.href, currentUrl);
+      if (fullUrl) {
+        break;
+      }
+      if (checkItem.href === '/cases') {
+        // if cases we need an equivalence to stop confusion in tab selection
+        continue;
+      }
+      // if the href partly matches, find the largest tab name for which the url partly matches
```

Comment:
I assume you mean the best match for the `href`, not `tab name`?

Reply:
yes, I can change that